src/manage/pep514utils.py

```diff
@@ -147,7 +147,69 @@ def _is_tag_managed(company_key, tag_name):
                 return True
         except FileNotFoundError:
             pass
-    return False
+        if not creating:
+            return False
+
+        # gh-11: Clean up invalid entries from other installers
+        # It's highly likely that our old MSI installer wouldn't properly remove
+        # its registry keys on uninstall, so we'll check for the InstallPath
+        # subkey and if it's missing, back up the key and then use it ourselves.
+        try:
+            with _reg_open(tag, "InstallPath") as subkey:
+                if subkey:
+                    # if InstallPath refers to a directory that exists,
+                    # leave it alone.
+                    p = winreg.QueryValueEx(subkey, None)[0]
+                    if p and Path(p).exists():
+                        return False
+        except FileNotFoundError:
+            pass
+        except OSError:
+            # If we couldn't access it for some reason, leave it alone.
+            return False
+
+    # Existing key is almost certainly not valid, so let's rename it,
+    # warn the user, and then continue.
+    LOGGER.debug("Key %s appears invalid, so moving it and taking it for this "
+                 "new install", tag_name)
+    try:
+        from _native import reg_rename_key
+    except ImportError:
+        LOGGER.debug("Failed to import reg_rename_key", exc_info=True)
+        return False
+
+    parent_name, _, orig_name = tag_name.replace("/", "\\").rpartition("\\")
+    with _reg_open(company_key, parent_name, writable=True) as tag:
+        if not tag:
+            # Key is no longer there, so we can use it
+            return True
+        for i in range(1000):
+            try:
+                new_name = f"{orig_name}.{i}"
+                # Raises standard PermissionError (5) if new_name exists
+                reg_rename_key(tag.handle, orig_name, new_name)
+                LOGGER.warn("An existing registry key for %s was renamed to %s "
+                            "because it appeared to be invalid. If this is "
+                            "correct, the registry key can be safely deleted. "
+                            "To avoid this in future, ensure that the "
+                            "InstallPath key refers to a valid path.",
+                            tag_name, new_name)
+                break
+            except FileNotFoundError:
+                LOGGER.debug("Original key disappeared, so we will claim it")
+                return True
+            except PermissionError:
+                LOGGER.debug("Failed to rename %s to %s", orig_name, new_name,
+                             exc_info=True)
```

Comment:
Should this clause break the loop?

Reply:
No, this is the most likely "failure" case (due to the new name already existing), so we want it to loop around. I thought I had a comment for that?

Reply:
Yes, comment is up at the `reg_rename_key` call, but I'll add another one here so nobody else has to ask the question.

```suggestion
            except PermissionError:
                LOGGER.debug("Failed to rename %s to %s", orig_name, new_name,
                             exc_info=True)
                # Continue, hopefully the next new_name is available
```